### PR TITLE
US114859 - More cleanup of d2l-all-courses and d2l-my-courses-container

### PR DESCRIPTION
**Pull request by svanherk**
Lots of random fixes here to try to bring consistency:
- Update the format of `d2l-my-courses` to class based
- Stop using Polymer's `listen`, delay hookup of event listeners with `AfterNextRender`
- `d2l-all-courses` wasn't really using the entity mixin as intended, switched to `entityFactory` to act more like the rest of the components
- Remove unused properties, mark private properties as private, add missing properties

`d2l-my-courses-content` will be tackled next PR, this is just going back over files I've started cleaning.

### CODE REVIEW

File: src/d2l-all-courses.js [hunk 1]
```diff
@@ -115,23 +110,23 @@ class AllCourses extends mixinBehaviors([
 				type: Boolean,
 				value: false
 			},
+			// Token JWT Token for brightspace | a function that returns a JWT token for brightspace
+			token: String,
+			// Initial search action, should combine with _enrollmentsSearchAction
+			enrollmentsSearchAction: Object,
```
Comment (AllanKerr) [inline src/d2l-all-courses.js:116]: Am I correct in that this was always being passed in and used, it was just never declared as a property?

Comment (svanherk) [reply src/d2l-all-courses.js:116]: Yup exactly, and then there's `_enrollmentsSearchAction` which is different.  I want to combine them, but this pass is just trying to document the current state of properties.

File: src/d2l-all-courses.js [hunk 2]
```diff
@@ -398,31 +387,37 @@ class AllCourses extends mixinBehaviors([
...
 
-	detached() {
-		this.unlisten(this.$.sortDropdown, 'd2l-menu-item-change', '_onSortOrderChanged');
-		this.unlisten(this.$.filterDropdownContent, 'd2l-dropdown-open', '_onFilterDropdownOpen');
-		this.unlisten(this.$.filterDropdownContent, 'd2l-dropdown-close', '_onFilterDropdownClose');
-		this.unlisten(this.$.filterMenu, 'd2l-filter-menu-change', '_onFilterChanged');
-		this.unlisten(this.$['search-widget'], 'd2l-search-widget-results-changed', '_onSearchResultsChanged');
+	disconnectedCallback() {
+		super.disconnectedCallback();
 		document.body.removeEventListener('set-course-image', this._onSetCourseImage.bind(this));
```
Comment (AllanKerr) [inline src/d2l-all-courses.js:418]: You didn't change it but isn't this remove broken since its being passed in a different function object?

Comment (svanherk) [reply src/d2l-all-courses.js:418]: Oh shoot, I missed updating that.  Nice catch!  I was just fixing this up in `d2l-my-courses-content-behavior` but didn't see this one.

**Comment by svanherk** [comment]
FYI @m6jones

**Pull request merged**